dagobah/core/core.py

```diff
@@ -690,6 +733,26 @@ def _serialize(self, include_run_logs=False, strict_json=False):
             result = json.loads(json.dumps(result, cls=StrictJSONEncoder))
         return result
 
+
+    def _implements_function(obj, functions):
+        """ Checks for the existence of a method or list of methods """
+        for expected_method in functions:
+            if not (hasattr(obj, expected_method) and
+                    callable(getattr(obj, expected_method))):
+                return False
+        return True
+
+
+    def implements_expandable(self, obj):
+        """ Checks for methods required to expand a task """
+        return self._implements_function(obj, 'expand')
```

Comment:
I'm not sure how this is working right now, it seems like it should throw an error since you're passing 3 arguments (including the implicit self) to a function which only takes two.